### PR TITLE
Introduce `HammerPointState` for new feature of combination

### PR DESCRIPTION
**Pull request by sonohoshi**
SSIA

```
public class HammerPointState : IState
{
    public Address Address { get; }
    public int RecipeId { get; }
    public int HammerPoint { get; private set; }
}
```

### CODE REVIEW

File: Lib9c/Model/State/HammerPointState.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            return List.Empty
+                .Add(ItemId.Serialize())
+                .Add(HammerPoint.Serialize());
+        }
+
+        public void UpdateHammerPoint(int point)
+        {
+            HammerPoint += point;
+        }
```
Comment (boscohyun) [inline Lib9c/Model/State/HammerPointState.cs:36]: How about rename this like `AddHammerPoint` or `IncreaseHammerPoint`?

File: .Lib9c.Tests/Model/State/HammerPointStateTest.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
+﻿namespace Lib9c.Tests.Model.State
+{
+    using System.Collections.Generic;
+    using Bencodex.Types;
+    using Libplanet;
+    using Libplanet.Crypto;
+    using Nekoyume.Model.State;
+    using Xunit;
+
+    public class HammerPointStateTest
```
Comment (boscohyun) [inline .Lib9c.Tests/Model/State/HammerPointStateTest.cs:10]: Can you please add a test that use `UpdateHammerPoint()` method?

File: Lib9c/Model/State/HammerPointState.cs (hunk 1 above)
Comment (sonohoshi) [reply Lib9c/Model/State/HammerPointState.cs:36]: I like 'Update' word but understand 'Increase' is better than it. I will change it~

File: .Lib9c.Tests/Model/State/HammerPointStateTest.cs (hunk 2 above)
Comment (sonohoshi) [reply .Lib9c.Tests/Model/State/HammerPointStateTest.cs:10]: OK I'll add that. THX

**Review by boscohyun — APPROVED** [review]

**Pull request merged**